src/admin/UserList.tsx

```diff
@@ -49,186 +41,159 @@ const columns: Column<User>[] = [
-          }
-        },
-        {
-          name: "Edit",
-          bulk: false,
-          triggers: ["click"],
-          call({ makeRequest }, data: User) {
-            return makeRequest<User>("get", data)
-              .then(user => {
-                dispatch({ type: 'set edit user', data: user });
-              });
-          }
-        },
-        {
-          name: "Delete",
-          bulk: false,
-          call({ makeRequest, executeAction }, data: User) {
-            if (window.confirm(`You are deleting user ${data.username}. Are you sure you want to continue?`)) {
-              return auth.delete(`users/${data.username}/`)
-                .then(() => {
-                  executeAction("refresh");
-                  alert(`Successfully deleted user ${data.username}`);
-                  dispatch({ type: 'show edit user', data: false });
-                }, (err: string[] | string) => {
-                  alert(typeof err === "string" ? ERRORS[err] : err.map(e => ERRORS[e]));
-                })
-            }
+  // grab a reference to the form state here, or else it won't update properly
+  const formState = context.formState;
 
-            return Promise.reject();
-          }
-        }
-      ]}
-      bagRef={(bag) => {
-        tableBagRef.current = bag;
-      }}
-    />
+  return (
     <Modal
-      show={state.showEditUser}
-      onHide={() => { dispatch({ type: 'show edit user', data: false }); }}
-      dialogClassName="UserList_modal"
+      show={props.show}
+      onHide={() => {
+        props.setShow(false);
+      }}
+      size="lg"
     >
-      <Modal.Header closeButton>
-        <Modal.Title>Editing {state.currentUser?.username}</Modal.Title>
+      <Modal.Header>
+        <Modal.Title>{props.user ? "Edit User" : "Create User"}</Modal.Title>
       </Modal.Header>
       <Modal.Body>
-        <ProfileForm user={state.currentUser} ref={editUserRef} renderFooter={(isLoading, hasErrors) => {
-          setIsLoading(isLoading);
-          setEditUserErrors(hasErrors);
-          return <></>;
-        }} />
+        <ProfileFormConsumer
+          context={context}
+          formId="editUserForm"
+          user={props.user}
+          onSubmitSuccessful={() => {
+            props.refreshTable();
+          }}
+          hideSubmit
+        />
       </Modal.Body>
       <Modal.Footer>
-        <Button variant="outline-secondary" onClick={() => { dispatch({ type: 'show edit user', data: false }); }}>
+        <Button
+          variant="secondary"
+          onClick={() => {
+            props.setShow(false);
+          }}
+        >
           Close without saving
         </Button>
-        <Button variant="outline-danger" onClick={() => {
-          if (tableBagRef.current) {
-            tableBagRef.current.executeAction("Delete");
-          }
-        }}>Delete user</Button>
         <Button
           type="submit"
-          disabled={isLoading || editUserErrors}
-          className="ml-auto"
```

Comment:
Submit button should keep this class, so that the "Cancel" and "Save" actions are physically (and socially) distanced from each other.

Reply:
https://github.com/UWmathNEWS/slugline-web/pull/13/commits/37baf6c8a17f29cdfce3cf4aed9a1aa1a50270c3